src/api.ts

```diff
@@ -26,88 +32,118 @@ export interface QueryDeviceResult {
 
 function isQueryDeviceResult(result: any): result is QueryDeviceResult {
     return typeof result.bit0 === 'boolean' &&
-        typeof result.bit1 === 'boolean' && 
+        typeof result.bit1 === 'boolean' &&
         typeof result.last_update_time === 'string';
 }
 
 export async function QueryDevice(host: string, jwt: string, payload: QueryDevicePayload): Promise<QueryDeviceResult> {
-    const url = `https://${host}/v1/query_two_bits`;
-
-    const headers = {
-        Authorization: `Bearer ${jwt}`,
-        'Content-Type':'application/json'
-    }
-
-    const body: QueryDevicePayload = {
-        device_token: payload.device_token,
-        transaction_id: payload.transaction_id || uuid(),
-        timestamp: payload.timestamp || Date.now()
-    };
-
-    const response = await fetch(url, { method: 'POST', headers, body: JSON.stringify(body) });
-
-    if (response.status !== 200) {
-        throw new Error(`Device check api returned ${response.status}: ${await response.text()}`);
-    }
-
-    const responseBody = await response.json();
-
-    if (!isQueryDeviceResult(responseBody)) {
-        throw new Error('Failed to parse response from device check api');
+    let request;
+    let response;
+    try {
+        const url = `https://${host}/v1/query_two_bits`;
+
+        const headers = {
+            Authorization: `Bearer ${jwt}`,
+            'Content-Type': 'application/json'
+        }
+
+        const body: QueryDevicePayload = {
+            device_token: payload.device_token,
+            transaction_id: payload.transaction_id || uuid(),
+            timestamp: payload.timestamp || Date.now()
+        };
+
+        request = new Request(url, { method: 'POST', headers, body: JSON.stringify(body) })
+        response = await fetch(request);
+
+        if (response.status !== 200) {
+            throw new Error(`Device check api returned ${response.status}: ${await response.text()}`);
+        }
+
+        const responseBody = await response.json();
+
+        if (!isQueryDeviceResult(responseBody)) {
+            throw new Error('Failed to parse response from device check api');
+        }
+
+        return responseBody;
+    } catch (err) {
```

Comment:
personally i would have the try catch just around the fetch, so you aren't catching your own errors

Reply:
the benefit to this is that we catch the details of the request / response regardless of point of failure